comps/cores/mega/micro_service.py

```diff
@@ -31,8 +37,61 @@ def __init__(
         provider: Optional[str] = None,
         provider_endpoint: Optional[str] = None,
         use_remote_service: Optional[bool] = False,
+        methods: List[str] = ["POST"],
+        llm_endpoint: Optional[str] = None,
     ):
         """Init the microservice."""
+
+        # Make sure LLM/Microservice endpoint are ready
+        logger.info(f"Init microservice - {name}")
+        if llm_endpoint is not None:
```

Comment:
instead of just waiting for LLM microservice. Should we make it more general to wait for all dependency?  maybe just traverse through DAG and make sure next node available first?

Reply:
In fact it does wait for the other dependency if they are not LLM related. This is what we observed when we tested this:
1. Microservice will wait for LLMs to be up
2. Megaservice will wait for the microservice to be up

https://github.com/opea-project/GenAIComps/blob/b611385d4ce94ca5486e5109e80ba4e7f3c4424f/comps/cores/mega/micro_service.py#L66-L93